src/chapters/databases-part2/game-logic.rst

```diff
@@ -0,0 +1,161 @@
+   </section>
+
+``play()`` Summary
+------------------
+
+#. By typing the URL of the game page into the address bar, players can access
+   the board before hiding any mines. To prevent this, we redirect them back to
+   the home page if they arrive at the page through a ``GET`` request.
+#. We need to collect two pieces of information from the game board: which cell
+   the player selected and whether or not they clicked the *Flag Mine*
+   checkbox.
+
+   a. Each button on the board has the same name, and the value for each one
+      matches its cell coordinates (like ``A10``).
+   b. However, the value for the *New Game* button is the empty string.
+   c. The checkbox input has a different name from the buttons.
+   d. The syntax for requesting the button and checkbox data is slightly
+      different.
+
+      .. sourcecode:: Python
+         :linenos:
+
+         # Request button data from the form:
+         request.form['button_name']
+
+         # Request checkbox data:
+         request.form.get('checkbox_name')
+
+#. Even though each button has a defined ``value``, we still need to include
+   some server-side validation. Invalid input redirects the user back to the
+   home page.
+#. After validating the data, we call the ``check_guess()`` function to see if
+   the player clicked a mine or made a safe choice on the board.
+
+Update ``mines.html``
+---------------------
+
+As the user plays the game, the appearance of the board changes. There are 4
+possibilities for the cells in the table. Each one is either:
+
+#. **Unselected**: These appear with a gray background and have an active, blue
+   button.
+#. **Selected and safe**: These appear with a white background and display a
+   number showing how many mines are in the surrounding spaces.
+#. **Flagged**: These appear with a goldenrod background and have an active,
+   blue button.
+#. **Mined**: These appear with a red background and an inactive button.
+   Clicking a mined cell ends the game and reveals the locations of all
+   remaining mines on the board.
+
+The ``play()`` function calls ``check_guess()`` each time the user clicks a
+button on the board. The results determine the how that cell should change
+after it is selected.
+
+To finish our Minesweeper game, we need to update the code in ``mines.html``.
+Our goal is to properly display the board after each turn. Also, we'll provide
+different feedback messages to indicate victory or defeat.
+
+Open the ``mines.html`` template in Visual Studio Code, then code along with
+the following videos.
+
+Change Cells After Click
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube-nocookie.com/embed/an8sh-Xp5PQ" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
```

Comment:
So not a biggie, but from 1:20 to2:05 Line 59 = session.modified = True is seen.

It was moved to line 69 in the starter code

Reply:
Thanks.  I see the statement, and it is also present on line 70 in the video.  These are redundant, so I probably removed the first one after making the video.  The latter is hit regardless of how the conditionals turn out.

I vote we leave this video alone.  In a follow up clip (video 7a on the Gameplay Functions page), there's a deeper explanation of the code, and this one has the offending line 59 absent.